### PR TITLE
atc: Warn on unused resources vs erroring out, solves #4177

### PR DESCRIPTION
**Pull request by kallisti5**
fixes: #4177

### CODE REVIEW

**Comment by kallisti5** [comment]
Please check this one thoroughly. Builds but untested in applying a pipeline as of yet.

**Comment by kallisti5** [comment]
Test YAML:
```yaml
resources:
  - name: A-time-trigger
    type: time
    source:
      start: 1:00 AM
      stop: 6:00 AM
      location: America/Chicago
  - name: B-time-trigger
    type: time
    source:
      start: 1:00 AM
      stop: 6:00 AM
      days: [Monday]
      location: America/Chicago
jobs:
  - name: ((type))-triggered
    public: true
    plan:
      - get: ((type))-time-trigger
        trigger: true
      - task: echo ((type))
        config:
          platform: linux
          image_resource:
            type: registry-image
            source: { repository: busybox }
          run:
            path: /bin/sh
            args:
              - -c
              - echo 'hello mate'
```

applied with ```-v type=A``` or ```-v type=B```

**Comment by kallisti5** [comment]
TODO: The unit tests fail.  ```Describe("unused resources", func() {```

Since i'm changing the error to a warning (and I don't see unit tests for warnings) does the whole "unused resources" check just go away?

I'm also assuming "unused resources" are harmless with this code.

File: atc/validate.go [hunk 1]
```diff
@@ -175,21 +175,25 @@ func validateResourceTypes(c Config) error {
...
 	usedResources := usedResources(c)
 
-	var errorMessages []string
 	for _, resource := range c.Resources {
 		if _, used := usedResources[resource.Name]; !used {
-			message := fmt.Sprintf("resource '%s' is not used", resource.Name)
-			errorMessages = append(errorMessages, message)
+			warnings = append(warnings, ConfigWarning{
+				Type:    "resources",
+				Message: resource.Name + " : is not used in pipeline",
```
Comment (analytically) [inline atc/validate.go]: space before the colon makes it look strange

**Comment by deniseyu** [comment]
hey @kallisti5, I had a go at fixing these unit tests. The reason why the unit tests broke is because they were expecting to read off errors, which are no longer returned; they needed to be changed so they read from objects in the `warnings` slice. I also took some liberty of changing some wording around slightly and simplifying some of the tests - they were definitely a bit more confusing than they needed to be!

thanks for your work on this. I couldn't push directly to your branch so I've added my changes as a patch file. After unzipping this file, you can move this file to your local Concourse repo and run `git apply apply.patch`.

[apply.patch.zip](https://github.com/concourse/concourse/files/3772626/apply.patch.zip)

**Comment by xtremerui** [comment]
hi @kallisti5 are you still working on this PR? Thx!

**Comment by kallisti5** [comment]
Nah.  I really don't have the bandwidth at the moment got them.   Consider these abandoned (but nothing technically wrong with the idea.. just no time on my end to implement)

**Comment by xtremerui** [comment]
Consider there is a patch file that might already working and the issue is valid, if you don't mind @kallisti5 could concourse team pick this up?

**Comment by xtremerui** [comment]
closing as of https://github.com/concourse/concourse/pull/5060

**Pull request closed without merging**